pkg/operators/ebpf/attach.go

```diff
@@ -145,6 +149,91 @@ func (i *ebpfInstance) attachProgram(gadgetCtx operators.GadgetContext, p *ebpf.
 		return link.AttachLSM(link.LSMOptions{
 			Program: prog,
 		})
+	case ebpf.PerfEvent:
+		perfType := uint32(unix.PERF_TYPE_SOFTWARE)
+		perfConfig := uint64(unix.PERF_COUNT_SW_CPU_CLOCK)
+		perfSampleType := uint64(unix.PERF_SAMPLE_RAW)
+		frequency := uint64(0)
+		name, ok := strings.CutPrefix(p.SectionName, perfEventPrefix)
+		// allow overriding parameters; TODO: discuss and add more? there's A LOT of options to cover
+		if ok {
+			if tmp := i.config.GetString("programs." + name + ".perf.type"); tmp != "" {
+				switch tmp {
+				case "software":
+					perfType = unix.PERF_TYPE_SOFTWARE
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 32)
+					if err != nil {
+						return nil, fmt.Errorf("unknown perf type %q for program %q", tmp, name)
+					}
+					perfType = uint32(val)
+				}
+			}
+			if tmp := i.config.GetString("programs." + name + ".perf.config"); tmp != "" {
+				switch tmp {
+				case "count_sw_cpu_clock":
+					perfConfig = unix.PERF_COUNT_SW_CPU_CLOCK
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 64)
+					if err != nil {
+						return nil, fmt.Errorf("unknown perf config %q for program %q", tmp, name)
+					}
+					perfConfig = val
+				}
+			}
+			if tmp := i.config.GetString("programs." + name + ".perf.sampleType"); tmp != "" {
+				switch tmp {
+				case "sample_raw":
+					perfSampleType = unix.PERF_SAMPLE_RAW
+				default:
+					// Try to get raw value // TODO: should we support it in this way as well?
+					val, err := strconv.ParseUint(tmp, 10, 64)
+					if err != nil {
+						return nil, fmt.Errorf("unknown perf config %q for program %q", tmp, name)
+					}
+					perfSampleType = val
+				}
+			}
+			if tmpFrequency := i.config.GetString("programs." + name + ".sampler.frequency"); tmpFrequency != "" {
+				var err error
+				frequency, err = strconv.ParseUint(tmpFrequency, 10, 64)
+				if err != nil {
+					return nil, fmt.Errorf("parsing frequency %q for program %q: %w", tmpFrequency, name, err)
+				}
+			}
+		}
+		for cpu := 0; cpu < runtime.NumCPU(); cpu++ {
+			fd, err := unix.PerfEventOpen(
+				&unix.PerfEventAttr{
+					Type:        perfType,
+					Config:      perfConfig,
+					Sample_type: perfSampleType,
+					Sample:      frequency,
+					Bits:        1 << 10,
```

Comment:
Define as a constant, as it is done in pkg/gadgets/profile/cpu/tracer/tracer.go:
```
const (
        // In C, struct perf_event_attr has a freq field which is a bit in a
        // 64-length bitfield.
        // In Golang, there is a Bits field which 64 bits long.
        // From C, we can deduce freq (which permits using frequency not period)
        // is the 10th bit.
        frequencyBit = 1 << 10
)
```

Reply:
Yeah, I was making it configurable as well, but then stopped to discuss here 😄 If we keep it fixed, I'll revert it to using the constant w/ explanation ofc.